### ISSUE TITLE
Could Deserialize<T> in BsonMapper.Deserialize be made public?

### ISSUE THREAD

**Issue by tetious** [mbdavid/LiteDB#895]
It is useful in some corner cases to be able to use configured custom mappers automatically when working with a BsonValue directly. 

`Deserialize<T>` seems to be intended for that, and even has a docstring, which may mean that it was previously public. Maybe it was made internal for some reason?

Could it be made public?

**Comment by mbdavid** [comment]
Hi @tetious, I don't remember (and don't found on git) if this method was public anytime. I try always use summary in all method (not all, but almost all).

As public method you have `ToObject<T>()` (if you are working with documents) or even `BsonValue.RawValue` if you are working with values only. But, if you is not enough to your case, I think it's possible make public... I always avoid offer public method to be easy to change if I need.

**Comment by tetious** [comment]
I'm doing something a bit weird, so I do seem to need Deserialize<T> and likely a non-generic counterpart, because I'm not working with a whole document at once, but a subset of properties (as BsonValues).

I've built an object diff implementation based on a BsonDocument that is a subset containing only the changes between two POCOs. Then I wrote a reflection based patcher that takes a POCO instance and applies the diff. The idea is that I'll be storing a stream of diffs that I can interrogate and replay as a syncing mechanism for keeping multiple copies of a database in sync. 

Anyway, I think I was overthinking things and just decided to use my diff object directly with ToObject. So I'm going to close this.

Thanks for the help!
